src/fr/animesama/src/eu/kanade/tachiyomi/extension/fr/animesama/AnimeSama.kt

```diff
@@ -73,84 +75,175 @@ class AnimeSama : ParsedHttpSource() {
+        var boolSyn = false
+        var boolGenre = false
+
+        document.select("#sousBlocMilieu > div").first()!!.children().forEach { element ->
+            if (boolSyn) {
+                boolSyn = false
+                description = element.text()
+            }
+
+            if (element.text() == "Synopsis") {
+                boolSyn = true
+            }
+
+            if (boolGenre) {
+                boolGenre = false
+                genre = element.text()
+            }
+
+            if (element.text() == "Genres") {
+                boolGenre = true
+            }
+        }
+        title = document.select("#titreOeuvre").text()
+        thumbnail_url = document.select("#coverOeuvre").attr("src")
     }
 
     // Chapters
     override fun chapterListSelector() = throw UnsupportedOperationException()
 
     override fun chapterFromElement(element: Element): SChapter = throw UnsupportedOperationException()
 
-    override fun chapterListParse(response: Response): List<SChapter> {
-        val document = response.asJsoup()
-        val javascriptUrl = document.select("body > script:nth-child(3)").attr("abs:src")
+    fun String.containsMultipleTimes(search: String): Boolean {
+        if (this.contains(search)) {
+            var temp = this.split(search)
+            return (temp.size > 2)
+        }
+        return false
+    }
 
+    override fun chapterListParse(response: Response): List<SChapter> {
+        var document = response.asJsoup()
+        val goodRequest = GET(document.baseUri() + "/scan/vf")
+        val goodResponse = client.newCall(goodRequest).execute()
+        document = goodResponse.asJsoup()
+
+        Log.v("AnimeSama", document.baseUri())
+        val javascriptUrl = "episodes.js?title=" + document.select("#titreOeuvre").text()
+        Log.v("AnimeSama", javascriptUrl)
         val newHeaders = headersBuilder()
             .add("Accept-Version", "v1")
             .build()
 
-        val request = GET(javascriptUrl, newHeaders)
+        val request = GET(document.baseUri() + "/" + javascriptUrl, newHeaders)
         val responsejs = client.newCall(request).execute()
-        val jsonDataString = responsejs.body.string()
+        var jsonDataString = responsejs.body.string()
 
-        return jsonDataString
+        val toJson = jsonDataString
             .split(" ", ",")
             .filter { it.contains("eps") && !it.contains("drive.google.com") }
             .mapNotNull { it.replace("=", "").replace("eps", "").toIntOrNull() }
             .sorted()
             .map { chapter ->
-                SChapter.create().apply {
-                    name = "Chapitre $chapter"
-                    setUrlWithoutDomain(javascriptUrl + "/$chapter")
+            }.asReversed()
+        var chapList: MutableList<SChapter> = ArrayList()
+        var retard = 0
+
+        Log.v("AnimeSama", document.html())
+        if (document.html().containsMultipleTimes("resetListe()")) {
```

Comment:
use css selectors like `script:containsData(resetListe)`, you can test for number of elements

Reply:
There is only one element who is like that
```javascript
/*
resetListe(); toujours au debut en premier
creerListe(debut, fin);newSP(special); on en met autant qu'on veut
finirListe(debut de la fin);
*/
resetListe();
creerListe(1, 8);newSP(8.2);
creerListe(9, 14);newSP(15.1);newSP(15.2);newSP(15.3);
creerListe(16, 44);newSP(44.5);newSP(44.6);
finirListe(45);
```
Comments are always present but the "real script" content isn't always present...
I will try using that and regex, i will do it tomorrow.